### PR TITLE
feat(query): show grants support mget table/db name by table/db id.

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Speed up show grants.

Under debug mode:
role1 owns 10 thounds table:
```
$ bendsql --query='show grants for role role1' --time                                                                                                                                          
28.317
```
After optimize, double Speed bonus
```
$ bendsql --query='show grants for role role1' --time                                                                                                                                          
15.020
```

Under release :

```
$ bendsql --query='show grants for role role1' --time                                                                                                                                                                                      
2.635
```
After optimize,
```
$ bendsql --query='show grants for role role1' --time                                                                                                                                                                                       
2.166
```

In this pr add two api in schema_api:

mget_database_names_by_ids;
mget_table_names_by_ids;

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [x] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [x] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14862)

### CODE REVIEW

**Review by SkyFan2002 — APPROVED** [review]

File: src/meta/api/src/schema_api_impl.rs [hunk 1]
```diff
@@ -2431,6 +2431,38 @@ impl<KV: kvapi::KVApi<Error = MetaError> + ?Sized> SchemaApi for KV {
...
+        // Batch get all table-name by id
+        let seq_names = self.mget_kv(&kv_keys).await?;
+        let mut table_names = Vec::with_capacity(kv_keys.len());
+
+        for (i, seq_name_opt) in seq_names.iter().enumerate() {
+            if let Some(seq_name) = seq_name_opt {
+                let name_ident: DBIdTableName = deserialize_struct(&seq_name.data)?;
+                table_names.push(name_ident.table_name);
+            } else {
+                debug!(
```
Comment (BohuTANG) [inline src/meta/api/src/schema_api_impl.rs:2457]: It's not meaningfu to add a `debug!` in `else` branch, when to use it?

Comment (TCeason) [reply src/meta/api/src/schema_api_impl.rs:2457]: A user can be granted by immuteable db or table and muteable db.

like this:

grant select on system.* to a;
grant update on default.* to a;

Under muteable catalog can not find system db'id.

**Review by drmingdrmer — COMMENTED** [review]
> Reviewed 1 of 16 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14862)* status: 1 of 16 files reviewed, 7 unresolved discussions (waiting on @BohuTANG, @everpcpc, @flaneur2020, and @TCeason)

___
*[`src/meta/api/src/schema_api.rs` line 212 at r1](https://reviewable.io/reviews/datafuselabs/databend/14862#-NsMnzyL2780-Lpdk4fg:-NsMnzyL2780-Lpdk4fh:bv2210s) ([raw file](https://github.com/datafuselabs/databend/blob/56b1e55f9a104f6c9846911aa8c69f7c3a97386d/src/meta/api/src/schema_api.rs#L212)):*
> ```XML
>     ) -> Result<(TableIdent, Arc<TableMeta>), KVAppError>;
> 
>     async fn list_tables_name_by_id(
> ```

It is a `mget` operation. Not a `list`.

`mget` gets values by keys. `list` gets values by a common prefix.
___
*[`src/meta/api/src/schema_api.rs` line 218 at r1](https://reviewable.io/reviews/datafuselabs/databend/14862#-NsMo8vU7PZax73Q3idG:-NsMo8vU7PZax73Q3idH:bup7k9o) ([raw file](https://github.com/datafuselabs/databend/blob/56b1e55f9a104f6c9846911aa8c69f7c3a97386d/src/meta/api/src/schema_api.rs#L218)):*
> ```XML
> 
>     async fn get_table_name_by_id(&self, table_id: MetaId) -> Result<String, KVAppError>;
>     async fn list_dbs_name_by_id(&self, db_id: Vec<MetaId>) -> Result<Vec<String>, KVAppError>;
> ```

It is `mget` not `list`.
___
*[`src/query/catalog/src/catalog/interface.rs` line 195 at r1](https://reviewable.io/reviews/datafuselabs/databend/14862#-NsMp-bq0W_sOo589hoN:-NsMp-bq0W_sOo589hoO:b-1c5d7j) ([raw file](https://github.com/datafuselabs/databend/blob/56b1e55f9a104f6c9846911aa8c69f7c3a97386d/src/query/catalog/src/catalog/interface.rs#L195)):*
> ```XML
>     async fn list_tables_name_by_id(
>         &self,
>         table_id: Vec<MetaId>,
> ```

It should be `table_ids`:



_Suggestion:_
```XML
        table_ids: Vec<MetaId>,
```
___
*[`src/query/catalog/src/catalog/interface.rs` line 204 at r1](https://reviewable.io/reviews/datafuselabs/databend/14862#-NsMp1ze3wDrKVxOEPYu:-NsMp1ze3wDrKVxOEPYv:b-e0qk5u) ([raw file](https://github.com/datafuselabs/databend/blob/56b1e55f9a104f6c9846911aa8c69f7c3a97386d/src/query/catalog/src/catalog/interface.rs#L204)):*
> ```XML
>     async fn list_dbs_name_by_id(
>         &self,
>         db_id: Vec<MetaId>,
> ```

`db_ids`
___
*[`src/query/catalog/src/catalog/session_catalog.rs` line 250 at r1](https://reviewable.io/reviews/datafuselabs/databend/14862#-NsMp4yL1WCXZwdSpLX2:-NsMp4yL1WCXZwdSpLX3:beqgdys) ([raw file](https://github.com/datafuselabs/databend/blob/56b1e55f9a104f6c9846911aa8c69f7c3a97386d/src/query/catalog/src/catalog/session_catalog.rs#L250)):*
> ```XML
>     async fn list_tables_name_by_id(
>         &self,
>         table_id: Vec<MetaId>,
> ```

`table_ids`
___
*[`src/query/catalog/src/catalog/session_catalog.rs` line 263 at r1](https://reviewable.io/reviews/datafuselabs/databend/14862#-NsMp6LMEBXVI4qg4Ntq:-NsMp6LMEBXVI4qg4Ntr:b-e0qk5u) ([raw file](https://github.com/datafuselabs/databend/blob/56b1e55f9a104f6c9846911aa8c69f7c3a97386d/src/query/catalog/src/catalog/session_catalog.rs#L263)):*
> ```XML
>     async fn list_dbs_name_by_id(
>         &self,
>         db_id: Vec<MetaId>,
> ```

`db_ids`

File: src/meta/api/src/schema_api_impl.rs (hunk 1 above)
Comment (BohuTANG) [reply src/meta/api/src/schema_api_impl.rs:2457]: Suggestions remove this debug, it's no use on production, and less use for development.

Comment (TCeason) [reply src/meta/api/src/schema_api_impl.rs:2457]: Ok. Comment is better?

Comment (TCeason) [reply src/meta/api/src/schema_api_impl.rs]: Fixed.

**Review by drmingdrmer — COMMENTED** [review]
> Reviewed 1 of 16 files at r1, 1 of 12 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14862)* status: 3 of 16 files reviewed, 5 unresolved discussions (waiting on @BohuTANG, @everpcpc, @flaneur2020, and @TCeason)

___
*[`src/meta/api/src/schema_api.rs` line 212 at r3](https://reviewable.io/reviews/datafuselabs/databend/14862#-NsNBK8k8akxLjNjbZ54:-NsNBK8k8akxLjNjbZ55:undefined) ([raw file](https://github.com/datafuselabs/databend/blob/58f8a764b7ee69333385e1b641edc8a6bd42e8fb/src/meta/api/src/schema_api.rs#L212)):*
> ```XML
>     ) -> Result<(TableIdent, Arc<TableMeta>), KVAppError>;
> 
>     async fn mget_tables_name_by_id(
> ```





_Suggestion:_
```XML
    async fn mget_table_names_by_id(
```
___
*[`src/meta/api/src/schema_api.rs` line 218 at r3](https://reviewable.io/reviews/datafuselabs/databend/14862#-NsNBPBG8fmkpP02y7pG:-NsNBPBG8fmkpP02y7pH:undefined) ([raw file](https://github.com/datafuselabs/databend/blob/58f8a764b7ee69333385e1b641edc8a6bd42e8fb/src/meta/api/src/schema_api.rs#L218)):*
> ```XML
> 
>     async fn get_table_name_by_id(&self, table_id: MetaId) -> Result<String, KVAppError>;
>     async fn mget_dbs_name_by_id(&self, db_ids: Vec<MetaId>) -> Result<Vec<String>, KVAppError>;
> ```





_Suggestion:_
```XML
    async fn mget_database_names_by_id(&self, db_ids: Vec<MetaId>) -> Result<Vec<String>, KVAppError>;
```
___
*[`src/query/catalog/src/catalog/session_catalog.rs` line 252 at r3](https://reviewable.io/reviews/datafuselabs/databend/14862#-NsNBmwV8XlDxXdW5O1K:-NsNBmwV8XlDxXdW5O1L:undefined) ([raw file](https://github.com/datafuselabs/databend/blob/58f8a764b7ee69333385e1b641edc8a6bd42e8fb/src/query/catalog/src/catalog/session_catalog.rs#L252)):*
> ```XML
>         table_ids: Vec<MetaId>,
>     ) -> databend_common_exception::Result<Vec<String>> {
>         self.inner.mget_tables_name_by_id(table_ids).await
> ```





_Suggestion:_
```XML
mget_table_names_by_id
```
___
*[`src/query/catalog/src/catalog/session_catalog.rs` line 261 at r3](https://reviewable.io/reviews/datafuselabs/databend/14862#-NsNBjk2BwzrejuEaWoO:-NsNBjk2BwzrejuEaWoP:undefined) ([raw file](https://github.com/datafuselabs/databend/blob/58f8a764b7ee69333385e1b641edc8a6bd42e8fb/src/query/catalog/src/catalog/session_catalog.rs#L261)):*
> ```XML
> 
>     // List the dbs name by meta ids.
>     async fn mget_dbs_name_by_id(
> ```





_Suggestion:_
```XML
mget_database_names_by_id
```

**Comment by TCeason** [comment]
> ```
> mget_dbs_name_by_id
> ```

Done

**Review by drmingdrmer — COMMENTED** [review]
> Reviewed 6 of 12 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14862)* status: 8 of 16 files reviewed, 2 unresolved discussions (waiting on @BohuTANG, @everpcpc, @flaneur2020, and @TCeason)

___
*[`src/query/catalog/src/catalog/session_catalog.rs` line 261 at r3](https://reviewable.io/reviews/datafuselabs/databend/14862#-NsNBjk2BwzrejuEaWoO:-NsNlGLW1cE3y0KlS88v:b-ejjqhk) ([raw file](https://github.com/datafuselabs/databend/blob/58f8a764b7ee69333385e1b641edc8a6bd42e8fb/src/query/catalog/src/catalog/session_catalog.rs#L261)):*
> ```XML
> 
>     // List the dbs name by meta ids.
>     async fn mget_dbs_name_by_id(
> ```

This  method name should be updated too.

**Comment by TCeason** [comment]
> > Reviewed 6 of 12 files at r4, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14862)_ status: 8 of 16 files reviewed, 2 unresolved discussions (waiting on @BohuTANG, @everpcpc, @flaneur2020, and @TCeason)
> 
> _[`src/query/catalog/src/catalog/session_catalog.rs` line 261 at r3](https://reviewable.io/reviews/datafuselabs/databend/14862#-NsNBjk2BwzrejuEaWoO:-NsNlGLW1cE3y0KlS88v:b-ejjqhk) ([raw file](https://github.com/datafuselabs/databend/blob/58f8a764b7ee69333385e1b641edc8a6bd42e8fb/src/query/catalog/src/catalog/session_catalog.rs#L261)):_
> 
> > ```
> >     // List the dbs name by meta ids.
> >     async fn mget_dbs_name_by_id(
> > ```
> 
> This method name should be updated too.

Already modify in the lastest commit.

File: src/query/service/src/catalogs/default/database_catalog.rs [hunk 2]
```diff
@@ -296,6 +296,20 @@ impl Catalog for DatabaseCatalog {
         }
     }
 
+    #[async_backtrace::framed]
+    async fn mget_table_names_by_id(&self, table_id: Vec<MetaId>) -> Result<Vec<String>> {
```
Comment (sundy-li) [inline src/query/service/src/catalogs/default/database_catalog.rs:300]: by_ids?

Comment (sundy-li) [reply src/query/service/src/catalogs/default/database_catalog.rs:300]: table_ids

File: src/meta/api/src/schema_api_impl.rs [hunk 3]
```diff
@@ -2431,6 +2431,32 @@ impl<KV: kvapi::KVApi<Error = MetaError> + ?Sized> SchemaApi for KV {
         Ok(table_name.unwrap().table_name)
     }
 
+    #[logcall::logcall("debug")]
+    #[minitrace::trace]
+    async fn mget_table_names_by_ids(
+        &self,
+        table_ids: Vec<MetaId>,
```
Comment (sundy-li) [inline src/meta/api/src/schema_api_impl.rs:2438]: why no pass by slice?

**Review by drmingdrmer — COMMENTED** [review]
> Reviewed 2 of 16 files at r1, 5 of 12 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14862)* status: 9 of 16 files reviewed, 4 unresolved discussions (waiting on @BohuTANG, @everpcpc, @flaneur2020, and @TCeason)

___
*[`src/query/catalog/src/catalog/session_catalog.rs` line 261 at r3](https://reviewable.io/reviews/datafuselabs/databend/14862#-NsNBjk2BwzrejuEaWoO:-NsQeWIF8JrAaPZ5qYFF:b-fl2f8s) ([raw file](https://github.com/datafuselabs/databend/blob/58f8a764b7ee69333385e1b641edc8a6bd42e8fb/src/query/catalog/src/catalog/session_catalog.rs#L261)):*
<details><summary><i>Previously, TCeason wrote…</i></summary><blockquote>

Already modify in the lastest commit.
</blockquote></details>

No. They are not modified :(

**Comment by TCeason** [comment]
> mget_dbs_name_by_id

O, I got, mget_database_names_by_ids

File: src/meta/api/src/schema_api_impl.rs (hunk 3 above)
Comment (TCeason) [reply src/meta/api/src/schema_api_impl.rs]: Done

**Review by drmingdrmer — APPROVED** [review]
The meta-data part LGTM.

> Reviewed 11 of 12 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14862#-:-NsQk1xcC4f7BTXhoXmv:byjipnz)* status: 15 of 16 files reviewed, 4 unresolved discussions (waiting on @BohuTANG, @everpcpc, @flaneur2020, @sundy-li, and @TCeason)

**Review by drmingdrmer — APPROVED** [review]
> Reviewed 3 of 3 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14862)* status: 15 of 16 files reviewed, 4 unresolved discussions (waiting on @BohuTANG, @everpcpc, @flaneur2020, @sundy-li, and @TCeason)

**Pull request merged**